src/debug.ts

```diff
@@ -43,20 +47,65 @@ export async function makeDebugConfigurations(ctx: FolderContext) {
     }
 }
 
-// Return array of DebugConfigurations based on what is in Package.swift
-function createDebugConfigurations(ctx: FolderContext): vscode.DebugConfiguration[] {
+// Return array of DebugConfigurations for executables based on what is in Package.swift
+function createExecutableConfigurations(ctx: FolderContext): vscode.DebugConfiguration[] {
     const executableProducts = ctx.swiftPackage.executableProducts;
 
-    return executableProducts.map((product) => {
-        return {
-            type: "lldb",
-            request: "launch",
-            name: `Run ${product.name}`,
-            program: "${workspaceFolder}/.build/debug/" + product.name,
-            args: [],
-            cwd: "${workspaceFolder}",
-            preLaunchTask: `swift: Build ${product.name}`
-        };    
-    
+    return executableProducts.flatMap((product) => {
+        return [
+            {
+                type: "lldb",
+                request: "launch",
+                name: `Debug ${product.name}`,
+                program: "${workspaceFolder}/.build/debug/" + product.name,
+                args: [],
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build Debug ${product.name}`
+            },
+            {
+                type: "lldb",
+                request: "launch",
+                name: `Release ${product.name}`,
+                program: "${workspaceFolder}/.build/release/" + product.name,
+                args: [],
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build Release ${product.name}`
+            }
+        ];    
     });
 }
+
+// Return array of DebugConfigurations for tests based on what is in Package.swift
+async function createTestConfigurations(ctx: FolderContext): Promise<vscode.DebugConfiguration[]> {
+    if (ctx.swiftPackage.getTargets('test').length === 0) { return []; }
+    try {
+        // If running on darwin. Find xctest exe and run pointing at xctest resources
```

Comment:
You _should_ be able to build and run on Darwin without Xcode by just installing the toolchain. It doesn't work with tests at the moment but we probably shouldn't lock ourselves into requiring Xcode

Reply:
Once we have custom path for swift I can look into that. The default at the moment is use Xcode. Im not sure how xctest stuff works with additional tool chains.

Reply:
@0xTim I've have a further investigation and I'm not sure you can run tests without Xcode being installed. The `xctest` executable doesn't come with the swift toolchain. It is only available from a Xcode installation.

Reply:
Ah I was hoping that had been fixed. I've opened this as an SR bug https://bugs.swift.org/browse/SR-15020 a while back so lets stick with what we have for now until it's been fixed